### ISSUE TITLE
triplet loss with softmax

### ISSUE THREAD

**Issue by muna-cs** [KaiyangZhou/deep-person-reid#307]
I have some confusion about using triplet and softmax

1- if I choose (loss=triplet) and (sampler=RandomSampler) for training this meaning triplet with softmax?
I read [here](https://github.com/KaiyangZhou/deep-person-reid/issues/155#issuecomment-484363594_) when the sampler=RandomIdentitySampler and loss=triplet this meaning triplet with softmax, but I used sampler=RandomSampler with triplet and I tried it once with label smoothing and once without, I found the result changes but if this setting (sampler=RandomSampler, loss=triplet) is meaning only triplet but softmax is off,  so why the results are changed with and without label smoothing??
2- this line for triplet with softmax
 loss = self.weight_t * loss_t + self.weight_x * loss_x
but the default value for self.weight_x=0
is this meaning the softmax(cross-entropy) is off?

Thank you

**Comment by KaiyangZhou** [comment]
to use triplet loss, you need to set `RandomIdentitySampler` so each identity will have multiple images within one minibatch

tune `weight_x ` to select a proper weight for softmax when you want to combine triplet with softmax (see table 12 in https://arxiv.org/pdf/1905.00953.pdf)

**Comment by muna-cs** [comment]
it is clear, thank you
but in label smoothing:
how does the label smoothing modify the final result even if the softmax is off ( weight_x = 0 ) ?

**Comment by KaiyangZhou** [comment]
can you show the full commands or more details about your experiments?
